tools/infrasys/src/root.rs

```diff
@@ -0,0 +1,196 @@
+use super::{error, KeyRole, Result};
+use log::{trace, warn};
+use pubsys_config::SigningKeyConfig;
+use rusoto_core::Region;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::process::Command;
+
+/// The tuftool macro wraps Command to simplify calls to tuftool, adding region functionality.
+macro_rules! tuftool {
+    ($region:expr, $format_str:expr, $($format_arg:expr),*) => {
+        let arg_str = format!($format_str, $($format_arg),*);
+        trace!("tuftool arg string: {}", arg_str);
+        let args = shell_words::split(&arg_str).context(error::CommandSplit { command: &arg_str })?;
+        trace!("tuftool split args: {:#?}", args);
+
+        let status = Command::new("tuftool")
+            .args(args)
+            .env("AWS_REGION", $region)
+            .status()
+            .context(error::TuftoolSpawn)?;
+
+        ensure!(status.success(), error::TuftoolResult {
+            command: arg_str,
+            code: status.code().map(|i| i.to_string()).unwrap_or_else(|| "<unknown>".to_string())
+        });
+    }
+}
+
+pub fn check_root(root_role_path: &Path) -> Result<()> {
+    if root_role_path.is_file() {
+        warn!("Please delete file at {}", root_role_path.display());
```

Comment:
Probably want to elaborate a bit more.
```suggestion
        warn!("Cowardly refusing to overwrite the existing root.json at {}. Please manually delete it and run again.", root_role_path.display());
```